mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -195,6 +243,12 @@ void mlir::populateArithToEmitCPatterns(TypeConverter &typeConverter,
     ArithOpConversion<arith::SubIOp, emitc::SubOp>,
     CmpFOpConversion,
     SelectOpConversion
-  >(typeConverter, ctx);
+  >(typeConverter, ctx)
+  .add<
+    CastOpConversion<arith::FPToSIOp, /*opHasTruncateSemantics=*/true>,
+    CastOpConversion<arith::FPToUIOp, /*opHasTruncateSemantics=*/true>,
+    CastOpConversion<arith::SIToFPOp, /*opHasTruncateSemantics=*/false>,
+    CastOpConversion<arith::UIToFPOp, /*opHasTruncateSemantics=*/false>
```

Comment:
I think it might be nicer to split the (u|s)int to fp and fp to (u|s)int conversions. Especially when we add code that lowers using a manual truncation, the code diverges more for the two kinds of conversions.

Reply:
Done, there's a bit of duplication now but reading is easier without a template.